design-proposals/klip-20_remove_terminate.md

```diff
@@ -0,0 +1,121 @@
+source and then stop each of these in turn, via a `TERMINATE` statement, before finally being able 
+to drop the stream or table. 
+
+From a script, it is not possible to determine the id of the queries that need to be stopped, and 
+those query ids will change from one run to another in most cases. Hence it is generally not 
+possible to drop a table or stream in a script.
+
+The problem is compounded by the fact that _mutiple_ persistent queries can be writing into the
+source or table.  However, [KLIP-17](klip-17-sql-union.md) proposes to correct this, so that there
+will be, at-most, a single persistent query writing to any table or stream.
+
+There also seems to be no clear benefit of being able to terminate a query without dropping the 
+associated stream or table. Especially, given there is no means to restart it.
+
+Consider that `CSAS` and `CTAS` statements are equivalent to `MATERIALIZED VIEW`s used in RDBSs, 
+(as proposed in [KLIP-19](klip-19-materialize-views.md)). In a traditional RDBS the underlying 
+process by which the materialized view is kept up to data is not exposed to the user.  There is
+no 'persistent query' to either list or terminate. Hence we propose we should remove the 
+problematic `TERMINATE` from our syntax and instead allow users to drop the view, even if there 
+are queries writing to it. If such queries exist they will simply be stop as part of the `DROP`.   
+
+## What is in scope
+
+* Removing `TERMINATE` from our SQL syntax.
+* Allowing streams and tables to be dropped while there are active queries writing to them.
+* Automatically stopping any persistent query that write to a stream or table when that stream or table is dropped.
+
+## What is not in scope
+
+* Allowing streams and tables to be dropped while they are used as a source to downstream views. 
+This check will remain, as it is needed to ensure downstream views don't break. 
+In the future, we may support a `DROP TABLE CASCADE` sytnax to help drop a hierarchy of MVs.
+* Removal of `show queries` and displaying query ids in `DESCRIBE`. Until such time that we can automatically 
+recover failed persistent queries we should continue to expose query state to users.
+In time, with automated recovery, it may be possible to completely hide this complexity from users.
+
+## Value/Return
+
+Remove `TERMINATE` will make KSQL easier to operate, which will help drive adoption.
+
+## Public APIS
+
+* `TERMINATE` will be removed from our SQL syntax.
+* `DROP STREAM|TABLE` will allow users to drop sources that still have active upstream queries writing to them.
+
+
+## Design
+
+This KLIP could potentially be done independently of [KLIP-17](klip-17-sql-union.md), which restores 
+the (1 - 0..1) relationship between source and persistent query. However, this is likely more complex 
+as there are potentially multiple queries writing to a source. This may make recovering from the 
+command topic more challenging.
```

Comment:
I'm not sure I understand why this makes recovery more difficult? If I recall correctly, we don't start any of the queries until we finish creating all of the topologies in recovery mode (see `CommandRunner:L154`)

Reply:
If KLIP-17 is done, then dropping a stream/table only needs to check that the stream/table is not the source of another query.  Without KLIP-17 it must also check there are no `INSERT INTO` writing to the stream/table.